### PR TITLE
fix(kafka): default_acl

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

fixes `default_acl` field incorrect behavior during import operation

## Why this way

to deliver quality product

### CODE REVIEW

**Comment by Serpentiel** [comment]
@dewan-ahmed could you please check if this resolves your issue?

**Comment by dewan-ahmed** [comment]
Hi @Serpentiel thanks for this change. I tested with the dev version of the provider. With the new change, terraform plan is attempting a force replacement and wants to recreate the resource. I don’t think this is what we intend to do. I have shared a screen-recording of the terraform plan command with you.

File: internal/service/kafka/kafka.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func aivenKafkaSchema() map[string]*schema.Schema {
 	aivenKafkaSchema["default_acl"] = &schema.Schema{
 		Type:        schema.TypeBool,
 		Optional:    true,
+		ForceNew:    true,
```
Comment (ivan-savciuc) [inline internal/service/kafka/kafka.go:28]: Why do we force recreation of the resource here? I think this filed only makes sense when during resource creation

File: internal/service/kafka/kafka.go [hunk 2]
```diff
@@ -226,5 +227,12 @@ func resourceKafkaRead(ctx context.Context, d *schema.ResourceData, m interface{
 		})
 	}
 
+	if d.IsNewResource() {
```
Comment (ivan-savciuc) [inline internal/service/kafka/kafka.go:230]: I think it is up to the user to set this value. What is the reason why we set it to false here?

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]
Please explain what was the reason for this change?

File: internal/service/kafka/kafka.go (hunk 2 above)
Comment (Serpentiel) [reply internal/service/kafka/kafka.go:230]: it works incorrectly during import

File: internal/service/kafka/kafka.go (hunk 1 above)
Comment (Serpentiel) [reply internal/service/kafka/kafka.go:28]: we cannot have this field edited, editing it should require a resource recreation

Comment (ivan-savciuc) [reply internal/service/kafka/kafka.go:28]: Can we make it `Computed` after creation?  Or make it use `CreateOnlyDiffSuppressFunc`?

File: internal/service/kafka/kafka.go (hunk 2 above)
Comment (ivan-savciuc) [reply internal/service/kafka/kafka.go]: ok ok, now i get it, I think we should make it computed when imported or basically in any other case other then creation.

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Comment by dewan-ahmed** [comment]
Thanks for reviewing @ivan-savciuc and thanks @Serpentiel for the change.

Adding a note here that I retested and the change is working as expected for `terraform import` 🎉

**Pull request merged**